### ISSUE TITLE
Talisker uses proxy address instead of client address for private endpoint filtering

### ISSUE THREAD

**Issue by maxiberta** [canonical-ols/talisker#91]
Clients can access any private endpoint by spoofing the proxy header:

```bash
$ curl https://myservice/_status/metrics -H 'X-Forwarded-For: 127.0.0.1'
# HELP some_metric
# TYPE some_metric counter
some_metric 1.0
```
I'm hitting taliskers's private endpoints behind a reverse proxy (which correctly sets in this case `X-Forwarded-For: 127.0.0.1 <client.ip>`). `TALISKER_NETWORKS` is not set; though I believe that by setting `TALISKER_NETWORKS=<internal.network>` all private endpoints will be accessible by remote clients for this reason.

**Comment by bloodearnest** [comment]
I could have sworn I'd already made this change. I certainly hit this issue already.

It's a tricky problem. The FEs *must* strip X-Forwarded-For for it to be secure in any way. And really, access to those specific urls should be controlled at the FEs. But we have the app-level check in place to prevent accidental expose by default.

**Comment by bloodearnest** [comment]
Fixed in #92
